File: Spring-Boot/ci-cd/core/src/main/java/com/deploy/core/hello/controller/HelloController.java
package com.deploy.core.hello.controller;

import com.deploy.core.hello.dto.HelloRequestDto;
import com.deploy.core.hello.dto.HelloResponseDto;
import org.springframework.http.ResponseEntity;

public interface HelloController {
    ResponseEntity<HelloResponseDto> create(HelloRequestDto dto);
    ResponseEntity<HelloResponseDto> findById(Long id);
}

File: Kotiln/missing-kotlin-parameter-exception/src/main/kotlin/core/missingkotlinparameterexception/controller/HelloController.kt
package core.missingkotlinparameterexception.controller

import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class HelloController {
    @PostMapping("/hello")
    fun hello(
        @RequestBody request: HelloRequest
    ): ResponseEntity<HelloRequest> {
        return ResponseEntity.ok(request)
    }
}

data class HelloRequest(
    val name: String
)
